diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index a71d296..f7e327d 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -49,7 +49,7 @@ if (WIN32)
     else (MINGW)
         list (APPEND QBT_APP_SOURCES ../qbittorrent.rc)
     endif (MINGW)
-    list(APPEND QBT_APP_SOURCES ../qbittorent.exe.manifest)
+    list(APPEND QBT_APP_SOURCES ../qbittorrent.exe.manifest)
 endif (WIN32)
 
 if (UNIX)

